### ISSUE TITLE
tB Resolves Some References Differently Than VB6

### ISSUE THREAD

**Issue by jpbro** [WaynePhillipsEA/twinbasic#757]
**Describe the bug**
tB resolves references to ambiguous names differently than VB6, which can break things.

**To Reproduce**
Consider this admittedly smelly code:

```
Module MainModule
    Private Type TestType
        TypeMember As Integer
        TypeMember2 As Integer
    End Type
     
    Public Sub Test(MyType As TestType)
        MyType.TypeMember = &H7fffffff&   ' Overflows in tB, but works in VB6
        MsgBox MyType.TypeMember
    End Sub
    
    Public Sub Main()
        Dim X As TestType
        Test X
    End Sub
End Module

Module OtherModule
    Public Type TestType
        TypeMember As Long
    End Type
End Module
```

There are 2 defined Types with the same name. One is Public in one module, and the other is Private in another module. They have different members, so they aren't fully interchangeable.

If you have a parameter that references the type in a public method of the module with the Private type definition, VB6 will use the Public definition from the other module. tB will use the more locally-scoped version, which can cause problems. 

If you try to disambiguate (incompletely) by using the following method signature:

`Public Sub Test(MyType As NewProject.TestType)`

tB will show the following errors:

```
'TestType' is ambiguous. could be:
   [Type] NewProject.MainModule.TestType
   [Type] NewProject.OtherModule.TestType
```

VB6 will use the Public definition without any errors.

If you fully disambiguate by using the following method signature:

`Public Sub Test(MyType As OtherModule.TestType)`

Everything works as expected.

**Expected behavior**
This is admittedly bad code that should be fixed, but for 100% back-compat tB should resolve the same as VB6.  Maybe this would be a good candidate for a warning so that we can be alerted to the code smell and fix it when migrating?

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC compiler version 0.14.31

**Additional context**

**Comment by wqweto** [comment]
Does `Private Sub Test2(MyType As TestType)` use the local private `TestType`?

This is quite obscure edge case for symbol resolution that I doubt it's ever described in VBx spec.

**Comment by jpbro** [comment]
@wqweto - I might have to rescind this report actually, as I've just tried the simplified version in VB6 again and I can't get it to behave as I described now. I either made a mistake, or there must be some other combination of factors that gets it to behave as described. 

In any case, @WaynePhillipsEA you should ignore this issue for now until I can get a better test put together to properly compare the tB/VB6 behaviour.

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.14.34, thanks!
